### DISCUSSION TITLE
What's the difference between access credentials and an authenticated HTTP client

### DISCUSSION THREAD

**Discussion by berkobob** [google/googleapis.dart#469] [Q&A] [answered]
What's the difference between access credentials and authenticated HTTP client? I need the Bearer token but that's available in both. It's more easily available via `AccessCredentials` but the `AuthClient` class seems to have more about auto refreshing which is important to me.

**Comment by kevmoo** [comment, marked as answer]
Good question!

`AccessCredentials` is easier if you just want the bearer token. It'd be pretty easy to copy the logic in the auto-refreshing client. It's all open source!

**Reply by berkobob** [reply]
So, to be clear, only `AuthClient` will auto-refresh? 
Just out of interest, what is the difference between them? Is there a doc you can point me to?

**Reply by kevmoo** [reply]
The docs are light on this repo.

the credential one just does a one-off auth and gives you the creds. That's it.
The other one is "smart". It tracks the TTL of the credentials and makes sure to refresh before making a request that would be with old creds. 🤷

**Comment by berkobob** [comment]
I've discovered another difference, `AccessCredentials` has a `.toJson()` method which means I can store the credentials in `SharedPreferences`. Is there any reason I can't store credentials there?

Finally, how do I use the refresh token in `AccessCredentials`? Do I need to do it manually with a `POST` request or is there a method I cannot find? Thanks.

**Reply by kevmoo** [reply]
You should create a refreshing client – or at least look at that source code – to understand how to use them.
